src/ripple/app/tx/Transaction.cpp

```diff
@@ -52,6 +52,30 @@ Transaction::Transaction (STTx::ref sit, Validate validate)
     }
 }
 
+Transaction::Transaction (STTx::ref sit, Validate validate, std::string& reason)
+    : mInLedger (0),
+      mStatus (INVALID),
+      mResult (temUNCERTAIN),
+      mTransaction (sit)
+{
+    try
+    {
+        mFromPubKey.setAccountPublic (mTransaction->getSigningPubKey ());
+        mTransactionID  = mTransaction->getTransactionID ();
+        mAccountFrom    = mTransaction->getSourceAccount ();
+    }
+    catch (...)
+    {
+        return;
```

Comment:
I'd prefer to remove the constructor as well. I left a comment that I think suggests how this could be accomplished fairly easily.